spec/requests/profile_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+describe "Profile", type: :request do
+  describe "#index" do
+    context "when user is not logged in" do
+      before { get profile_index_path, params: {uid: "user-id"} }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context "when user is logged in" do
+      let(:user1) { create(:user1) }
+      before { sign_in user1 }
+      it "renders index template" do
+        get profile_index_path, params: {uid: user1.uid}
+        expect(response).to render_template(:index)
+      end
+
+      context "when profile belongs to logged in user" do
+        it "assigns profile instance to be that of logged in user" do
+          get profile_index_path, params: {uid: user1.uid}
+          expect(assigns(:profile)).to eq(user1)
+        end
+
+        context "when user has no moments and strategies" do
+          it "assigns stories instance with an empty array" do
+            get profile_index_path, params: {uid: user1.uid}
+            expect(assigns(:stories)).to match_array([])
```

Comment:
I'd probably change the test name to `it "have no stories" do` or something of that sort and change `assigns(:stories)` to use the response.body and make sure it doesn't have a story or something since `assigns` is dependent on `rails-controller-testing`.

Reply:
I think what I'll do here is test that the `response.body` doesn't have the Stories section. The view template wont render that section if there aren't any stories.